pkg/exchange/okex/parse.go

```diff
@@ -391,16 +390,67 @@ func parseAccount(v []byte) (*okexapi.Account, error) {
 	return &accounts[0], nil
 }
 
-func parseOrder(v *fastjson.Value) ([]okexapi.OrderDetails, error) {
-	data := v.Get("data").MarshalTo(nil)
+type OrderTradeEvent struct {
+	okexapi.OrderDetail
+
+	Code          types.StrInt64        `json:"code"`
+	Msg           string                `json:"msg"`
+	AmendResult   string                `json:"amendResult"`
+	ExecutionType okexapi.LiquidityType `json:"execType"`
+	// FillFee last filled fee amount or rebate amount:
+	// Negative number represents the user transaction fee charged by the platform;
+	// Positive number represents rebate
+	FillFee fixedpoint.Value `json:"fillFee"`
+	// FillFeeCurrency last filled fee currency or rebate currency.
+	// It is fee currency when fillFee is less than 0; It is rebate currency when fillFee>=0.
+	FillFeeCurrency string `json:"fillFeeCcy"`
+	// FillNotionalUsd Filled notional value in USD of order
+	FillNotionalUsd fixedpoint.Value `json:"fillNotionalUsd"`
+	FillPnl         fixedpoint.Value `json:"fillPnl"`
+	// NotionalUsd Estimated national value in USD of order
+	NotionalUsd fixedpoint.Value `json:"notionalUsd"`
+	// ReqId Client Request ID as assigned by the client for order amendment. "" will be returned if there is no order amendment.
+	ReqId     string           `json:"reqId"`
+	LastPrice fixedpoint.Value `json:"lastPx"`
+	// QuickMgnType Quick Margin type, Only applicable to Quick Margin Mode of isolated margin
+	// manual, auto_borrow, auto_repay
+	QuickMgnType string `json:"quickMgnType"`
+	// AmendSource Source of the order amendation.
+	AmendSource string `json:"amendSource"`
+	// CancelSource Source of the order cancellation.
+	CancelSource string `json:"cancelSource"`
+
+	// Only applicable to options; return "" for other instrument types
+	FillPriceVolume string `json:"fillPxVol"`
+	FillPriceUsd    string `json:"fillPxUsd"`
+	FillMarkVolume  string `json:"fillMarkVol"`
+	FillFwdPrice    string `json:"fillFwdPx"`
+	FillMarkPrice   string `json:"fillMarkPx"`
+}
 
-	var orderDetails []okexapi.OrderDetails
-	err := json.Unmarshal(data, &orderDetails)
+func (o *OrderTradeEvent) toGlobalTrade() (types.Trade, error) {
+	side := toGlobalSide(o.Side)
+	tradeId, err := strconv.ParseUint(o.TradeId, 10, 64)
 	if err != nil {
-		return nil, err
+		return types.Trade{}, fmt.Errorf("unexpected trade id [%s] format: %w", o.TradeId, err)
 	}
-
-	return orderDetails, nil
+	return types.Trade{
+		ID:            tradeId,
+		OrderID:       uint64(o.OrderId),
+		Exchange:      types.ExchangeOKEx,
+		Price:         o.FillPrice,
+		Quantity:      o.FillSize,
+		QuoteQuantity: o.FillPrice.Mul(o.FillSize),
+		Symbol:        toGlobalSymbol(o.InstrumentID),
+		Side:          side,
+		IsBuyer:       side == types.SideTypeBuy,
+		IsMaker:       o.ExecutionType == okexapi.LiquidityTypeMaker,
+		Time:          types.Time(o.FillTime.Time()),
+		// charged by the platform is positive in our design, so added the `Neg()`.
+		Fee:           o.FillFee.Neg(),
```

Comment:
The fillFee default is currently negative because it is charged by the platform. However, in our design, the value should be positive, so I added the neg.

Reply:
If it's a rebate, then the result will be the opposite.